docs/data.rst

```diff
@@ -19,6 +19,26 @@ module. The salient functions from it are as follows:
 .. automodule:: hypothesis.strategies
   :members:
 
+.. _shrinking:
+
+~~~~~~~~~
+Shrinking
+~~~~~~~~~
+
+When using strategies it is worth thinking about how the data *shrinks*.
+Shrinking is the process by which Hypothesis tries to produce human readable
+examples when it finds a failure - it takes a complex example and turns it
+into a simpler one.
+
+Each strategy defines an order in which it shrinks - these aren't always
+straightforward, and are somewhat subject to change, but are worth being aware
```

Comment:
I think a clearer explanation of the importance of shrinking and of our compatibility guarantees would be useful here, eg:

> Each strategy defines an order in which examples shrink.  While you usually won't need to think about this, it can be useful to know when creating large or very complex strategies - shrinking in the "wrong" direction can reduce performance and have serious impacts on the quality of reported examples.
> Shrinking behaviour is fairly stable, but may occasionally change between minor versions of Hypothesis as we improve particular strategies or the underlying engine.